### PR TITLE
discovery-feed: Adapt to EknContent property type change

### PR DESCRIPTION
**Pull request by ptomato**
The EkncContentObjectModel:discovery-feed-content property is now a
JsonObject instead of a GVariant. We get the required node directly,
thereby avoiding another GVariant conversion step.

https://phabricator.endlessm.com/T20702

### CODE REVIEW

File: search-provider/eks-discovery-feed-provider.c [hunk 1]
```diff
@@ -223,19 +223,15 @@ get_day_of_year (void)
...
-select_string_from_variant_from_day (GVariant *variant)
+static const char *
+select_string_from_array_from_day (JsonArray *array)
 {
-  gsize size = g_variant_n_children (variant);
+  guint size = json_array_get_length (array);
   if (size == 0)
     return NULL;
 
   gint index = get_day_of_week () % size;
```
Comment (ebassi) [inline search-provider/eks-discovery-feed-provider.c]: Not part of the patch, but I'd strongly recommend using `idx` or `index_` instead of `index`, in this code. Yes, there's an `index()` function in the C standard library, declared in `string.h`. You should be seeing compiler warnings when building.

Comment (ptomato) [reply search-provider/eks-discovery-feed-provider.c]: I :heart: my `ix`
